### PR TITLE
All flags and Color migrated to value classes

### PR DESCRIPTION
**Pull request by zTrap**
Related to https://github.com/kordlib/kord/issues/711

### CODE REVIEW

File: common/src/commonMain/kotlin/Color.kt [hunk 1]
```diff
@@ -1,62 +1,55 @@
...
-
-        override fun deserialize(decoder: Decoder): Color = Color(decoder.decodeInt())
-
-        override fun serialize(encoder: Encoder, value: Color) {
-            encoder.encodeInt(value.rgb)
-        }
+        private val delegate = Int.serializer()
+        override val descriptor: SerialDescriptor = delegate.descriptor
+        override fun deserialize(decoder: Decoder): Color = Color(delegate.deserialize(decoder))
+        override fun serialize(encoder: Encoder, value: Color) = delegate.serialize(encoder, value.rgb)
```
Comment (DRSchlaubi) [inline common/src/commonMain/kotlin/Color.kt:53]: I think we no longer need a custom serializer now, right?

File: ksp-processors/src/main/kotlin/generation/bitflags/FactoryFunctions.kt [hunk 2]
```diff
@@ -45,7 +45,6 @@ internal fun FileSpec.Builder.addFactoryFunctions() {
         }
     }
     addFactoryFunctionForIterable(baseParameterType = entityCN, IterableType.VARARG)
-    addFactoryFunctionForIterable(baseParameterType = collectionCN, IterableType.VARARG)
```
Comment (DRSchlaubi) [inline ksp-processors/src/main/kotlin/generation/bitflags/FactoryFunctions.kt:48]: Why was this removed?

File: ksp-processors/src/main/kotlin/generation/bitflags/BitFlagsGeneration.kt [hunk 3]
```diff
@@ -79,9 +79,11 @@ internal fun BitFlags.generateFileSpec(originatingFile: KSFile) = fileSpecForGen
     }
     addClass(collectionCN) {
         addCollectionKDoc()
+        addAnnotation<JvmInline> {}
         addAnnotation<Serializable> {
             addMember("with·=·%T.Serializer::class", collectionCN)
         }
+        addModifiers(PUBLIC, VALUE)
```
Comment (DRSchlaubi) [inline ksp-processors/src/main/kotlin/generation/bitflags/BitFlagsGeneration.kt:86]: Doesn't kotlinpoet add the public modifier automatically?

File: ksp-processors/src/main/kotlin/generation/bitflags/BitFlagsGeneration.kt [hunk 4]
```diff
@@ -79,9 +79,11 @@ internal fun BitFlags.generateFileSpec(originatingFile: KSFile) = fileSpecForGen
     }
     addClass(collectionCN) {
         addCollectionKDoc()
+        addAnnotation<JvmInline> {}
         addAnnotation<Serializable> {
             addMember("with·=·%T.Serializer::class", collectionCN)
```
Comment (DRSchlaubi) [inline ksp-processors/src/main/kotlin/generation/bitflags/BitFlagsGeneration.kt:84]: I think we don't need a custom serializer here either?

**Review by DRSchlaubi — CHANGES_REQUESTED** [review]

File: ksp-processors/src/main/kotlin/generation/bitflags/FactoryFunctions.kt (hunk 2 above)
Comment (zTrap) [reply ksp-processors/src/main/kotlin/generation/bitflags/FactoryFunctions.kt:48]: Value classes can't be used as vararg 😞

File: common/src/commonMain/kotlin/Color.kt (hunk 1 above)
Comment (zTrap) [reply common/src/commonMain/kotlin/Color.kt:53]: Unfortunately, we need it here. We need to (de)serialize color without alpha, but backed value contains it (`packedRGB` prop). Alpha removes only by triggering `rgb` prop getter.

**Comment by lukellmann** [comment]
I'd like to discuss this some more before deciding what to do with this, see https://discord.com/channels/556525343595298817/1245773750486438060

**Comment by zTrap** [comment]
All changes moved to #886
